module/mempool/herocache/dns_cache.go

```diff
@@ -0,0 +1,143 @@
+package herocache
+
+import (
+	"net"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	herocache "github.com/onflow/flow-go/module/mempool/herocache/backdata"
+	"github.com/onflow/flow-go/module/mempool/herocache/backdata/heropool"
+	"github.com/onflow/flow-go/module/mempool/stdmap"
+)
+
+type DNSCache struct {
+	ipCache  *stdmap.Backend
+	txtCache *stdmap.Backend
+}
+
+func NewDNSCache(sizeLimit uint32, logger zerolog.Logger) *DNSCache {
+	return &DNSCache{
+		txtCache: stdmap.NewBackendWithBackData(
+			herocache.NewCache(sizeLimit, 8, heropool.LRUEjection, logger)),
+		ipCache: stdmap.NewBackendWithBackData(
+			herocache.NewCache(sizeLimit, 8, heropool.LRUEjection, logger)),
+	}
+}
+
+// PutIpDomain adds the given ip domain into the cache.
+func (d *DNSCache) PutIpDomain(domain string, timestamp int64, addresses []net.IPAddr) bool {
+	i := ipEntity{
+		id:        domainToIdentifier(domain),
+		domain:    domain,
+		addresses: addresses,
+		timestamp: timestamp,
+	}
+
+	return d.ipCache.Add(i)
+}
+
+// PutTxtDomain adds the given txt domain into the cache.
+func (d *DNSCache) PutTxtDomain(domain string, timestamp int64, addresses []string) bool {
```

Comment:
Consider renaming (potentially misleading) `addresses` to `TxtRecords` or `TextRecords` as per [1].

[1] https://en.wikipedia.org/wiki/TXT_record